Alignment/RPDataFormats/plugins/TotemRPIncludeAlignments.cc

```diff
@@ -0,0 +1,270 @@
+  {
+    for (RPAlignmentCorrectionsSequence::const_iterator iit = fit->begin(); iit != fit->end(); ++iit)
+    {
+      const TimeValidityInterval &tvi = iit->first;
+      const RPAlignmentCorrections *corr = & iit->second;
+
+      bounds[tvi.first].push_back( pair<bool, const RPAlignmentCorrections*>(true, corr) );
+
+      TimeValue_t delta = (tvi.last != TimeValidityInterval::EndOfTime()) ? (1ULL << 32) : 0;  // input resolution is 1s
+      bounds[tvi.last + delta].push_back( pair<bool, const RPAlignmentCorrections*>(false, corr) );
+    }
+  }
+  
+  // build correction sums per interval
+  set<const RPAlignmentCorrections*> accumulator;
+  RPAlignmentCorrectionsSequence result;
+  //  bool gap_found = false;
+  for (map< TimeValue_t, vector< pair<bool, const RPAlignmentCorrections*> > >::const_iterator tit = bounds.begin(); tit != bounds.end(); ++tit)
+  {
+    for (vector< pair<bool, const RPAlignmentCorrections*> >::const_iterator cit = tit->second.begin(); cit != tit->second.end(); ++cit)
+    {
+      bool add = cit->first;
+      const RPAlignmentCorrections *corr = cit->second;
+
+      if (add)
+        accumulator.insert(corr);
+      else 
+        accumulator.erase(corr);
+    }
+    
+    map< TimeValue_t, vector< pair<bool, const RPAlignmentCorrections*> > >::const_iterator tit_next = tit;
+    tit_next++;
+    if (tit_next == bounds.end())
+      break;
+
+    TimeValue_t delta = (tit_next->first != TimeValidityInterval::EndOfTime()) ? 1 : 0; // minimal step
+    TimeValidityInterval tvi(tit->first, tit_next->first - delta);
+
+    if (verbosity)
+    {
+      printf("\tfirst=%10s, last=%10s: alignment blocks=%li\n",
+        TimeValidityInterval::ValueToUNIXString(tvi.first).c_str(),
+        TimeValidityInterval::ValueToUNIXString(tvi.last).c_str(),
+        accumulator.size()
+      );
+    }
+
+    for (set<const RPAlignmentCorrections*>::iterator sit = accumulator.begin(); sit != accumulator.end(); ++sit)
+      result[tvi].AddCorrections(*(*sit));
+  }
+
+  return result;
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void TotemRPIncludeAlignments::PrepareSequence(const string &label, RPAlignmentCorrectionsSequence &seq, const vector<string> &files) const
+{
+  if (verbosity)
+    printf(">> TotemRPIncludeAlignments::PrepareSequence(%s)\n", label.c_str());
+
+  vector<RPAlignmentCorrectionsSequence> sequences;
+  for (unsigned int i = 0; i < files.size(); i++)
+    sequences.push_back(RPAlignmentCorrectionsSequence(files[i]));
+
+  seq = Merge(sequences);
+}
+
+//----------------------------------------------------------------------------------------------------
+
+std::auto_ptr<RPAlignmentCorrections> TotemRPIncludeAlignments::produceMeasured(const RPMeasuredAlignmentRecord &iRecord)
+{
+  std::auto_ptr<RPAlignmentCorrections> alignments(new RPAlignmentCorrections(acMeasured));
+  return alignments;
```

Comment:
I changed the code according to your instruction and hope this will solve the issue with compilation